monitoring/instance_manager_collector.go

```diff
@@ -0,0 +1,174 @@
+package monitoring
+
+import (
+	"strings"
+	"sync"
+
+	"github.com/sirupsen/logrus"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	metricsclientset "k8s.io/metrics/pkg/client/clientset/versioned"
+
+	"github.com/prometheus/client_golang/prometheus"
+
+	"github.com/longhorn/longhorn-manager/datastore"
+	"github.com/longhorn/longhorn-manager/types"
+)
+
+type InstanceManagerCollector struct {
+	*baseCollector
+
+	kubeMetricsClient *metricsclientset.Clientset
+	namespace         string
+
+	instanceManagerUsageCPUMetric      metricInfo
```

Comment:
How about flipping these and just using the below?

cpuUsageMetric`
`cpuRequestMetric`
`memoryUsageMetric`
`memoryRequestMetric`

Reply:
basically index by HW component then specific metric.

Reply:
Thank you! Fixed